src/cpu.cr

```diff
@@ -0,0 +1,135 @@
+module CPU
+  enum Vendor
+    Unknown,
+    Intel,
+    AMD
+  end
+
+  @[Flags]
+  enum Features : UInt64
+    SSE,
+    SSE2,
+    SSE3,
+    SSSE3,
+    SSE4_1,
+    SSE4_2,
+    AVX,
+    AVX2
+  end
+
+  @@vendor_string = ""
+  @@vendor = Vendor::Unknown
+  @@features = Features::None
+
+  def self.cpuid(fn : Int32, subfn : Int32 = 0)
+    buf = [0_u32, 0_u32, 0_u32, 0_u32]
```

Comment:
Aren't `StaticArray` bound to the stack? I supposed they could not be returned from methods.